chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala

```diff
@@ -314,27 +345,45 @@ trait ChainUnitTest
   }
 
   def createChainHandler(): Future[ChainHandler] = {
-    val genesisHeaderF = setupHeaderTableWithGenesisHeader()
+    val (chainHandler, genesisHeaderF) = setupHeaderTableWithGenesisHeader()
     genesisHeaderF.map(_ => chainHandler)
   }
 
   def withChainHandler(test: OneArgAsyncTest): FutureOutcome = {
     makeFixture(createChainHandler, destroyHeaderTable)(test)
   }
 
+  // Creates and populates BlockHeaderTable with block headers 562375 to 571375
+  def createPopulatedChainHandler(): Future[ChainHandler] = {
+    createPopulatedBlockHeaderDAO().flatMap { blockHeaderDAO =>
+      blockHeaderDAO.getAtHeight(FIRST_BLOCK_HEIGHT).map { blockHeaderVec =>
+        ChainHandler(Blockchain(blockHeaderVec, blockHeaderDAO))
+      }
+    }
+  }
+
+  def withPopulatedChainHandler(test: OneArgAsyncTest): FutureOutcome = {
+    makeFixture(createPopulatedChainHandler, destroyHeaderTable)(test)
+  }
+
   /** Creates the [[org.bitcoins.chain.models.BlockHeaderTable]] */
   private def setupHeaderTable(): Future[Unit] = {
     ChainDbManagement.createHeaderTable(dbConfig = dbConfig,
                                         createIfNotExists = true)
   }
 
   /** Creates the [[org.bitcoins.chain.models.BlockHeaderTable]] and inserts the genesis header */
-  private def setupHeaderTableWithGenesisHeader(): Future[BlockHeaderDb] = {
+  private def setupHeaderTableWithGenesisHeader(): (
+      ChainHandler,
```

Comment:
do you see this tuple expanding? If so we should just bite the bullet now and make a case class

Reply:
I don't but if any field ever gets added I'll change it to a case class